scraping/reddit/reddit_custom_scraper.py

```diff
@@ -0,0 +1,325 @@
+from scraping.scraper import ScrapeConfig, Scraper, ValidationResult
+import bittensor as bt
+from common.data import DataEntity, DataLabel, DataSource, DateRange
+from typing import List
+import asyncpraw
+from scraping.reddit.utils import (
+    is_valid_reddit_url,
+    validate_reddit_content,
+    get_time_input,
+    get_custom_sort_input,
+    normalize_label,
+)
+from scraping.reddit.model import RedditContent, RedditDataType
+import traceback
+import datetime as dt
+import asyncio
+import random
+import os
+
+
+class RedditCustomScraper(Scraper):
+    """
+    Scrapes Reddit data using the Reddit Scraper Lite actor.
+    """
+
+    USER_AGENT = "User-Agent: python:data-universe-app:v1: (by /u/Soft-Strawberry-4067)"
+
+    async def validate(self, entities: List[DataEntity]) -> List[ValidationResult]:
+        """Validate the correctness of a DataEntity by URI."""
+        if not entities:
+            return []
```

Comment:
It should be a bug for us to try to validate no items.

Should raise an exception or at least return a failed ValidationResult?

Reply:
Consistent across all scrapers. Will resolve in a follow up across them.

Reply:
SG. ty